pkg/operator/baremetal_test.go

```diff
@@ -71,3 +95,40 @@ func TestCreateMariadbPasswordSecret(t *testing.T) {
 		t.Logf("First Mariadb password is being preserved over re-creation as expected.")
 	}
 }
+
+func TestGetBaremetalProvisioningConfig(t *testing.T) {
+	u := &unstructured.Unstructured{Object: map[string]interface{}{}}
+	if err := yaml.Unmarshal([]byte(yamlContent), &u); err != nil {
+		t.Errorf("failed to unmarshall input yaml content:%v", err)
+	}
+	dynamicClient := fakedynamic.NewSimpleDynamicClient(runtime.NewScheme(), u)
+	baremetalConfig, err := getBaremetalProvisioningConfig(dynamicClient, "test")
+	if err != nil {
+		t.Logf("Unstructed Config:  %+v", u)
+		t.Fatalf("Failed to get Baremetal Provisioning Interface from CR %s", "test")
+	}
+	if baremetalConfig.ProvisioningInterface != expectedProvisioningInterface ||
+		baremetalConfig.ProvisioningIp != expectedProvisioningIP ||
+		baremetalConfig.ProvisioningNetworkCIDR != expectedProvisioningNetworkCIDR ||
+		baremetalConfig.ProvisioningDHCPExternal != expectedProvisioningDHCPExternal ||
+		baremetalConfig.ProvisioningDHCPRange != expectedProvisioningDHCPRange {
+		t.Logf("Actual BaremetalProvisioningConfig: %+v", baremetalConfig)
+		t.Logf("Expected : ProvisioningInterface: %s, ProvisioningIP: %s, ProvisioningNetworkCIDR: %s, ProvisioningDHCPExternal: %t, expectedProvisioningDHCPRange: %s", expectedProvisioningInterface, expectedProvisioningIP, expectedProvisioningNetworkCIDR, expectedProvisioningDHCPExternal, expectedProvisioningDHCPRange)
+		t.Fatalf("failed getBaremetalProvisioningConfig. One or more BaremetalProvisioningConfig items do not match the expected config.")
+	}
+}
+
+func TestGetIncorrectBaremetalProvisioningCR(t *testing.T) {
+	u := &unstructured.Unstructured{Object: map[string]interface{}{}}
+	if err := yaml.Unmarshal([]byte(yamlContent), &u); err != nil {
+		t.Errorf("failed to unmarshall input yaml content:%v", err)
+	}
+	dynamicClient := fakedynamic.NewSimpleDynamicClient(runtime.NewScheme(), u)
+	baremetalConfig, err := getBaremetalProvisioningConfig(dynamicClient, "test1")
+	if err != nil {
```

Comment:
wouldn't be the client expected to error here and so `getBaremetalProvisioningConfig`

For units we prefer to have one test per function i.e `TestGetBaremetalProvisioningConfig` and loop over multiple test cases within it. e.g https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/vsphere/machine_scope_test.go#L139-L261